### PR TITLE
Update on DQA documentation and bug fix

### PR DESCRIPTION
**Pull request by ganiyuolalekan**
@Sensei-akin, @Justus-coded, @AdesholaAfolabi 

Updated the DQA documentation so it is more descriptive.

Also found a bug in `preprocessing.preprocess`. There would always be a clash if any feature column has age contained within it.

### Error

```commandline
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-24-410f60d92945> in <module>
----> 1 pp.preprocess(dataset, target_column='SalePrice', project_path='example/')

~\anaconda3\lib\site-packages\slik_wrangler\preprocessing.py in preprocess(data, target_column, train, select_columns, display_inline, project_path, **kwargs)
   1199 
   1200     if display_inline:
-> 1201         _preprocess(data=data,target_column=target_column,train=train,select_columns=select_columns,\
   1202                display_inline=display_inline,project_path=project_path,**kwargs)
   1203     else:

~\anaconda3\lib\site-packages\slik_wrangler\preprocessing.py in _preprocess(data, target_column, train, select_columns, display_inline, project_path, **kwargs)
   1162     else:
   1163         PROCESSED_TEST_PATH = os.path.join(project_path, 'validation_data.pkl')
-> 1164         _preprocess_non_target_col(data=train_df,PROCESSED_DATA_PATH = PROCESSED_TEST_PATH,display_inline=display_inline,\
   1165                                  select_columns=select_columns,**kwargs)
   1166 

~\anaconda3\lib\site-packages\slik_wrangler\preprocessing.py in _preprocess_non_target_col(data, processed_data_path, display_inline, select_columns, **kwargs)
   1007                 match = re.search(r'(.*?)[Aa]ge.*', column).group()
   1008                 age_column = str(match)
-> 1009                 data = bin_age(data,age_column)
   1010 
   1011         for name, dtype in data.dtypes.iteritems():

~\anaconda3\lib\site-packages\slik_wrangler\preprocessing.py in bin_age(dataframe, age_col, add_prefix)
     52 
     53     bin_labels = ['Toddler/Baby', 'Child', 'Young Adult', 'Mid-Age', 'Elderly']
---> 54     data[prefix_name] = pd.cut(data[age_col], bins = [0,2,17,30,45,99], labels = bin_labels)
     55     data[prefix_name] = data[prefix_name].astype(str)
     56     return data

~\anaconda3\lib\site-packages\pandas\core\reshape\tile.py in cut(x, bins, right, labels, retbins, precision, include_lowest, duplicates, ordered)
    271             raise ValueError("bins must increase monotonically.")
    272 
--> 273     fac, bins = _bins_to_cuts(
    274         x,
    275         bins,

~\anaconda3\lib\site-packages\pandas\core\reshape\tile.py in _bins_to_cuts(x, bins, right, labels, precision, include_lowest, dtype, duplicates, ordered)
    405 
    406     side = "left" if right else "right"
--> 407     ids = ensure_int64(bins.searchsorted(x, side=side))
    408 
    409     if include_lowest:

TypeError: '<' not supported between instances of 'int' and 'str'
```

### Step to replicate

dataset: https://www.kaggle.com/competitions/house-prices-advanced-regression-techniques/data

```python
import re
from slik_wrangler.loadfile import read_file
from slik_wrangler import preprocessing as pp

dataset = read_file(train_file_path, index_col='Id')
pp.preprocess(dataset, target_column='SalePrice', project_path='.')
```

### Cause of Issue

```python
match = [c for c in dataset.columns if re.search(r'(.*?)[Aa]ge.*', c)]
print(match)
```

```commandline
['GarageType',
 'GarageYrBlt',
 'GarageFinish',
 'GarageCars',
 'GarageArea',
 'GarageQual',
 'GarageCond']
```

I wrote this all down because I wasn't the author of the function and don't know the effect of my fix.

### CODE REVIEW

File: slik_wrangler/dqa.py
Comment (AdesholaAfolabi) [inline slik_wrangler/dqa.py:152]: I think this is fine for sub-functions.
@Sensei-akin were you hoping for more descriptions here?

File: slik_wrangler/preprocessing.py
Comment (ganiyuolalekan) [reply slik_wrangler/preprocessing.py:1006]: Yeah, it does, to a degree. It ensures that the string ends with age and is not in between.

For instance, Customer Age and Age would work and GarageType and Age of Customers (instance) won't.

Comment (Sensei-akin) [reply slik_wrangler/preprocessing.py:1006]: This makes alot of sense. This is clearly a bug

**Comment by ganiyuolalekan** [comment]
@AdesholaAfolabi  @Sensei-akin @Justus-coded 

Just added the final bit for the `dqa` module

File: slik_wrangler/dqa.py
Comment (Sensei-akin) [reply slik_wrangler/dqa.py]: @AdesholaAfolabi For some of the modules. The doc strings are not descriptive

Comment (ganiyuolalekan) [reply slik_wrangler/dqa.py]: What models?

**Review by AdesholaAfolabi — APPROVED** [review]

**Comment by AdesholaAfolabi** [comment]
@Sensei-akin @Justus-coded we need at least two approvals before merging this PR. Please look at the changes and review/approve. Thanks.

**Review by Sensei-akin — APPROVED** [review]

**Comment by Sensei-akin** [comment]
@AdesholaAfolabi I have approved

File: slik_wrangler/dqa.py
Comment (ganiyuolalekan) [reply slik_wrangler/dqa.py]: Yeah it is!

**Pull request merged**
